crates/invoker-impl/src/invocation_task/mod.rs

```diff
@@ -489,6 +489,13 @@ fn service_protocol_version_to_header_value(
     }
 }
 
+fn invocation_id_to_header_value(invocation_id: &InvocationId) -> HeaderValue {
+    let value = Bytes::from(invocation_id.to_string());
+
+    HeaderValue::from_maybe_shared(value)
```

Comment:
you can just use the try_from impl on String, it will do the same as you, but guaranteed to use the version of the Bytes lib that the http library imports, which means definitely no copy